WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -1811,12 +1813,21 @@ private boolean addMediaVisualEditor(Uri uri, boolean isVideo) {
         MediaModel media = queueFileForUpload(uri, getContentResolver().getType(uri));
         MediaFile mediaFile = FluxCUtils.mediaFileFromMediaModel(media);
         if (media != null) {
-            mEditorFragment.appendMediaFile(mediaFile, path, mImageLoader);
+            appendMediaFileAndSavePost(mediaFile, path, mImageLoader);
         }
 
         return true;
     }
 
+    private void appendMediaFileAndSavePost(MediaFile mediaFile, String imageUrl, ImageLoader imageLoader) {
+        mEditorFragment.appendMediaFile(mediaFile, imageUrl, mImageLoader);
+        // after asking the Editor to append the Media File, save the Post object to the DB
+        // as we need to keep the new modifications made to the Post's body (that is, the inserted media)
+        if (updatePostObject()) {
+            savePostToDb();
```

Comment:
tl;dr good call, addressed in 15fb128
I hesitated a bit on this as it seemed harmful that I only wanted to get the latest post data and  then disptach a FLuxC event (which is done asynchronously), but then realized the `updatePostObject`  could take long if this is a big post. Nice catch!